extensions/2.0/Khronos/KHR_materials_coat/README.md

```diff
@@ -0,0 +1,361 @@
+
+### Parameter Combinations
+
+Values for coating properties can be defined using factors, textures, or both. If textures are not provided, respective texture components are assumed to have default values. All coating textures contain components in linear space except where noted. If both factors and textures are present, the factor value acts as a linear multiplier for the corresponding texture values.
+
+```
+coatWeight = coatFactor * coatTexture.r
+coatRoughness = coatRoughnessFactor * coatRoughnessTexture.g
+coatColor = coatColorFactor * sRGBToLinear(coatColorTexture.rgb)
+coatDarkening = coatDarkeningFactor
+coatAnisotropyStrength_final = coatAnisotropyStrength * coatAnisotropyTexture.b
+```
+
+In the case of anisotropy rotation, the combination of coatAnisotropyRotation and the rotation direction from coatAnisotropyTexture.rg is slightly more complex since one specifies an angle and the other a direction. We can convert the direction from the texture to an angle and then add to coatAnisotropyRotation.
+```
+coatAnisotropyRotation += atan(coatAnisotropyTexture.g, coatAnisotropyTexture.r);
+```
+
+## Basic Coating
+
+The coat effect is modeled via a microfacet BRDF. The BRDF is layered on top of the glTF 2.0 Metallic-Roughness material, excluding emission, using a new `fresnel_coat` function:
+
+```
+# base material (from glTF 2.0 Metallic-Roughness)
+material = mix(dielectric_brdf, metal_brdf, metallic)
+
+# coat layer
+coat_brdf = specular_brdf(
+  normal = coatNormal,
+  α = coatRoughness^2,
+  anisotropy = coatAnisotropy)
+
+# layering
+coated_material =
+  fresnel_coat(
+    normal = coatNormal,
+    ior = coatIor,
+    weight = coatWeight,
+    base = material * coatColor * coatDarkening,
+    layer = coat_brdf)
+```
+
+The `fresnel_coat` function adds a BRDF as a layer on top of another BSDF according to `weight` and a Fresnel term. The layer is weighted with `weight*fresnel(ior)`. The base is weighted with `1-(weight*fresnel(ior))`. `normal` is a float3 vector that affects the top layer, but not the base.
+
+The specular BRDF for the clearcoat layer `clearcoat_brdf` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#appendix-b-brdf-implementation).
+
+#### Layering
+
+The `fresnel_coat` function is computed using the Schlick Fresnel term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#appendix-b-brdf-implementation).
+
+```
+function fresnel_coat(normal, ior, weight, base, layer) {
+  f0 = ((1-ior)/(1+ior))^2
+  fr = f0 + (1 - f0)*(1 - abs(dot(V, normal)))^5
+  return mix(base, layer, weight * fr)
+}
+```
+
+#### Emission
+
+Emission is assumed to be on top of the coat layer in the layering stack. Consequently, the emission is not affected by coat.
```

Comment:
This is different than the original clearcoat extension where emission was underneath coat.
The intent in changing this is to make it more compatible with OpenPBR.